hw2 ex1 initial

igor-korost opened the pull request.

igor-korost committed:
hw2 ex1 initial

DmitryKhodakovsky commented on src/test/java/src/test/java/ru/training/at/hw2/ex1/SoftAssertsTest.java:
```diff
@@ -0,0 +1,163 @@
+    @Test
+    public void test() {
+        // Use WebDriverManager to manage browser driver
+        WebDriverManager.chromedriver().setup();
+        // Create new instance of WebDriver
+        webDriver = new ChromeDriver();
+        // implicit waits
+        webDriver.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);
+        // Start browser maximized
+        webDriver.manage().window().maximize();
```
Could be extracted to the `@BeforeMethod` of `BaseClass`

DmitryKhodakovsky commented on src/test/java/src/test/java/ru/training/at/hw2/ex1/SoftAssertsTest.java:
```diff
@@ -0,0 +1,163 @@
+        webDriver.manage().window().maximize();
+
+        // 1. Open test site by URL
+        String baseUrl = "https://jdi-testing.github.io/jdi-light/index.html";
```
Could be extracted to the constants in the base class

DmitryKhodakovsky commented on src/test/java/src/test/java/ru/training/at/hw2/ex1/SoftAssertsTest.java:
```diff
@@ -0,0 +1,163 @@
+        softAssert.assertTrue(isDisplayedNav);
+
+        // 6. Assert that there are 4 images on the Index Page and they are displayed
+        WebElement iconPractise = webDriver.findElement(By.cssSelector(".icon-practise"));
+        WebElement iconCustom = webDriver.findElement(By.cssSelector(".icon-custom"));
+        WebElement iconMulti = webDriver.findElement(By.cssSelector(".icon-multi"));
+        WebElement iconBase = webDriver.findElement(By.cssSelector(".icon-base"));
```
icons could be find as `List<webElement>` and through the loop could be checked for the `isDisplayer`

igor-korost replied:
Icons common parent (ancestor) is too far away to get them by findElements(). The code will be too cumbersome. And I don't know how to check whether I got the right icons in my List<WebElement>. For String Lists, I can simply print them with println().

DmitryKhodakovsky reviewed (changes requested):
Where is the ex2? 
Without ex2 work will not accepted

igor-korost committed:
hw2 ex1, ex2

-DriverManager methods extracted to the @BeforeMethod of BaseClass
-baseUrl extracted to the BaseClass
